File: frontend/cli/src/app/remitos/remito.ts
import { ArticulosPedido } from "../articulospedido/articulospedido";
import { Domicilio } from "../domicilios/domicilio";

export interface Remito {
  id: number;

  fechaArmado: Date;

  entregado: boolean;

  domicilio: Domicilio;

  total: number;

  articulosPedido: ArticulosPedido[];
}

File: backend/src/main/java/unpsjb/labprog/backend/model/Remito.java
package unpsjb.labprog.backend.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import java.util.Date;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.JoinColumn;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Remito {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;

    private Date fechaArmado;

    private boolean entregado;

    @ManyToOne
    @JoinColumn(name = "domicilio_id")
    private Domicilio domicilio;

}